jetstream/analysis.py

```diff
@@ -439,22 +476,64 @@ def run(self, current_date: datetime, dry_run: bool = False) -> None:
                 app_id=self._app_id_to_bigquery_dataset(self.config.experiment.app_id),
             )
 
-            metrics_table = self.calculate_metrics(exp, time_limits, period, dry_run)
+            metrics_dataframes = []
+            analysis_bases = []
+
+            for m in self.config.metrics[period]:
+                if isinstance(m.metric.analysis_basis, list):
+                    for analysis_basis in m.metric.analysis_basis:
+                        analysis_bases.append(analysis_basis)
+                else:
+                    analysis_bases.append(m.metric.analysis_basis)
+
+            analysis_bases = list(set(analysis_bases))
+
+            if len(analysis_bases) == 0:
+                continue
+
+            for analysis_basis in analysis_bases:
+                metrics_table = self.calculate_metrics(
+                    exp, time_limits, period, analysis_basis, dry_run
+                )
+
+                if dry_run:
+                    results.append(metrics_table)
+                else:
+                    metrics_dataframes.append(table_to_dataframe(metrics_table))
 
             if dry_run:
                 logger.info(
                     "Not calculating statistics %s (%s); dry run",
                     self.config.experiment.normandy_slug,
                     period.value,
                 )
-                results.append(metrics_table)
                 continue
 
-            metrics_data = table_to_dataframe(metrics_table)
+            segment_labels = [s.name for s in self.config.experiment.segments]
+            join_fields = [
+                "client_id",
+                "branch",
+                "enrollment_date",
+                "num_enrollment_events",
+                "analysis_window_start",
+                "analysis_window_end",
+                "exposure_date",
+                "num_exposure_events",
+            ] + segment_labels
+
+            metrics_data = reduce(
```

Comment:
I think this makes a single table that has both exposure- and enrollment-basis data in it. We need to calculate the statistics over the exposure-basis data and the enrollment-basis data separately. Keeping the tables separate might make more sense.

Anyway, if a metric has more than one basis won't the column names conflict?

Reply:
Indeed, the computations need to be done separately. Actually that simplifies things...